File: tester/harmony/react_native_openharmony/src/main/ets/RNOHCorePackage/turboModules/LogBoxTurboModule.ts
import { TurboModule, EventEmitter } from "../../RNOH/ts"


export class LogBoxTurboModule extends TurboModule {
  static NAME: Readonly<string> = "LogBox"

  public eventEmitter = new EventEmitter<{
    "SHOW": [],
    "HIDE": []
  }>()

  public show() {
    this.eventEmitter.emit("SHOW")
  }

  public hide() {
    this.eventEmitter.emit("HIDE")
  }
}

File: tester/harmony/react_native_openharmony/src/main/cpp/RNOHCorePackage/TurboModules/LogBoxTurboModule.cpp
#include "RNOHCorePackage/TurboModules/LogBoxTurboModule.h"

using namespace facebook;

namespace rnoh {
LogBoxTurboModule::LogBoxTurboModule(
    const ArkTSTurboModule::Context ctx,
    const std::string name)
    : ArkTSTurboModule(ctx, name) {
  methodMap_ = {
      ARK_METHOD_METADATA(show, 0),
      ARK_METHOD_METADATA(hide, 0),
  };
}
} // namespace rnoh
